cmd/csi-attacher/main.go

```diff
@@ -60,6 +61,7 @@ var (
 	timeout           = flag.Duration("timeout", 15*time.Second, "Timeout for waiting for attaching or detaching the volume.")
 
 	enableLeaderElection    = flag.Bool("leader-election", false, "Enable leader election.")
+	leaderElectionType      = flag.String("leader-election-type", "configmap", "the type of leader election, options are 'configmap' (default) or 'lease' (recommended)")
 	leaderElectionNamespace = flag.String("leader-election-namespace", "", "Namespace where this attacher runs.")
 	leaderElectionIdentity  = flag.String("leader-election-identity", "", "Unique identity of this attacher. Typically name of the pod where the attacher runs.")
```

Comment:
I don't think identity is needed anymore since the lib will use the hostname if it's not set. Should we deprecate the flag here now?

Reply:
Yes, please deprecate it (print a warning when it's used)

Reply:
@andrewsykim There's https://github.com/kubernetes-csi/csi-lib-utils/blob/master/deprecatedflags/deprecatedflags.go which can be used for deprecated flags.

Reply:
ah, thank you!